core/src/Platforms/iOS/iOSPlatformProxy.cs

```diff
@@ -120,6 +122,20 @@ public string GetProductName()
         {
             return _isMsal ? "MSAL.Xamarin.iOS" : "PCL.iOS";
         }
+        public string GetApplicationName()
+        {
+            return (NSString)NSBundle.MainBundle.InfoDictionary["CFBundleName"].ToString();
```

Comment:
Are all of these app name / app version strings guaranteed to have a value / not throw a null ref? Do you want to have a fallback ?

Reply:
You're right, added fallbacks to app name, version, and device id